arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Iterable.kt

```diff
@@ -758,17 +773,27 @@ public fun <A, B> Iterable<A>.align(b: Iterable<B>): List<Ior<A, B>> =
   this.align(b, ::identity)
 
 /**
- * aligns two structures and combine them with the given [Semigroup.combine]
+ * aligns two structures and combine them with [combine]
  */
 public fun <A> Iterable<A>.salign(
-  SG: Semigroup<A>,
+  combine: (A, A) -> A,
   other: Iterable<A>
-): Iterable<A> = SG.run {
+): Iterable<A> =
   align(other) {
     it.fold(::identity, ::identity) { a, b ->
-      a.combine(b)
+      combine(a, b)
     }
   }
+
+/**
+ * aligns two structures and combine them with the given [Semigroup.combine]
+ */
+@Deprecated(SemigroupDeprecation, ReplaceWith("SG.run { salign({ a, b -> a.combine(b) }, other) }"))
```

Comment:
I just tried to follow the pattern of having the `run` method always outside. But I'm happy to change it if it may bring problems.

Reply:
I've added a `SG::combine` extension method to improve most `ReplaceWith`, and moved `run { }` inside where needed. Having it outside results in issues with `ReplaceWith` since inside `run` the `this` context changes from `Iterable<A>` to `Semigroup` and thus `salign` is called on the wrong receiver.